### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by YegorVolkov**
Regarding successfull workflow reference :
Since docker/helm jobs have main branch restriction & I can not merge to main branch without pull request - only python CI job's workflow is checked. (here below)

---

PLease find below screenshot evidence of the branch protection in force:

---

Besides this - please find below screenshot of  local forked branch pull request where we can see `required` tag on python-ci job, and blocked merge option untill changes approved.

---

Please find below screenshot evidence of the staging environment deployment approval policy in force:

### CODE REVIEW

**Review by YuriiSmolii — CHANGES_REQUESTED** [review]
Please remove main branch protection and test whole pipeline run. We need it to validate task.

File: .github/workflows/reusable-deployment.yml
Comment (YuriiSmolii) [inline .github/workflows/reusable-deployment.yml]: This values shouldn't be hardcoded

**Review by YuriiSmolii — CHANGES_REQUESTED** [review]
The problem is if the user manually choose type and py version - the pipeline should deploy only the choosen artifact.

File: .github/workflows/reusable-deployment.yml
Comment (YegorVolkov) [reply .github/workflows/reusable-deployment.yml]: @YuriiSmolii 
Good Day!
Please find below link for successfull workflow with branch pull request approve NOT requested:
https://github.com/YegorVolkov/devops_todolist_cicd_task_6_polish_pipeline/actions/runs/10228145227

Please find below screenshot showing the commit hashes prooving that no code changes were done compared to the state of the `current dev branch pull_request` 
*in order to avoid creating new pull request on different branch which would require deleting of current fork, with doing all the task again


Please find below screenshot showing that staging env deployment is waiting for approval
image



----
*** **just noticed your comment for the values hard-coded** 
**- kindly ask you to refuse the pull-request so I can ask for review once rectified**
**thank you for patience**

**Pull request closed without merging**
